test/0_staking.js

```diff
@@ -204,6 +205,66 @@ describe('Candidates make stakes on themselves', () => {
         }
+
+        // Test moving of stakes
+        console.log('**** One of delegators moves stakes to another candidate');
+        let candidate_rec = constants.CANDIDATES[2].staking;
+        let delegator = delegators[0];
+
+        // initial stake on the initial candidate
+        let iStake = await StakingAuRa.instance.methods.stakeAmount(candidate, delegator).call();
+        let iStakeBN = new BN(iStake.toString());
+        // initial stake on the second candidate
+        let iStake_rec = await StakingAuRa.instance.methods.stakeAmount(candidate_rec, delegator).call();
+        let iStake_recBN = new BN(iStake_rec.toString());
+
+        let tx = await SnS(web3, {
+            from: delegator,
+            to: StakingAuRa.address,
+            method: StakingAuRa.instance.methods.moveStake(candidate, candidate_rec, minStakeBN.toString()),
+            gasPrice: '1000000000'
+        });
+        expect(tx.status, `Tx to move stake failed: ${tx.transactionHash}`).to.equal(true);
+
+        // final stake on the initial candidate (should have decreased)
+        let fStake = await StakingAuRa.instance.methods.stakeAmount(candidate, delegator).call();
+        let fStakeBN = new BN(fStake.toString());
+        let dStakeBN = fStakeBN.sub(iStakeBN);
+        expect(dStakeBN, `Stake on initial candidate ${candidate} didn't decrease`).to.be.bignumber.equal(minStakeBN.neg()); // x.neg() == -x
+
+        // final stake on the second candidate (should have increased)
+        let fStake_rec = await StakingAuRa.instance.methods.stakeAmount(candidate_rec, delegator).call();
+        let fStake_recBN = new BN(fStake_rec.toString());
+        let dStake_recBN = fStake_recBN.sub(iStake_recBN);
+        expect(dStake_recBN, `Stake on second candidate ${candidate_rec} didn't increase`).to.be.bignumber.equal(minStakeBN);
+
+        console.log('**** Moving stake must fail if delegator tries to move stake to the same candidate');
+        try {
+            let tx2 = await SnS(web3, {
+                from: delegator,
+                to: StakingAuRa.address,
+                method: StakingAuRa.instance.methods.moveStake(candidate, candidate, minStakeBN.toString()),
+                gasPrice: '1000000000'
+            });
+            expect(false, `Tx didn't throw an exception: ${tx2.transactionHash}. Tx status: ${tx2.status}`).to.equal(true);
+        }
+        catch (e) {
+            expect(e && e.toString() == REVERT_EXCEPTION_MSG, `Tx threw an unexpected exception: ` + e.toString()).to.equal(true)
+        }
+
+        console.log('**** Delegator can\'t move more staking tokens than one has');
+        try {
+            let tx3 = await SnS(web3, {
+                from: delegator,
+                to: StakingAuRa.address,
+                method: StakingAuRa.instance.methods.moveStake(candidate, candidate_rec, ((new BN(newNativeBalance)).add(new BN('1'))).toString()),
```

Comment:
Here it's enough to try to `moveStake(candidate, candidate_rec, minStakeBN.toString())` because we know that the `delegator` has already moved their one and only `minStakeBN` from the `candidate` pool.